### PR TITLE
[Feature Request] Posts and Events features and Spam mitigation issued fixed

### LINKED ISSUE #357: [Feature Request] Posts and Events features and Spam mitigation
**Is your feature request related to a problem? Please describe.**
Currently, In talawa-admin, the posts and events list contains hard-coded values and while creating events and posts some fields are missing. Also, as the count of posts and events increases, there is a need for pagination and search functionality.

Also, there is no way in Talawa-admin to detect the user who spammed in the organization group chat and block that user.

**Describe the solution you'd like**

* The posts and events list will be redesigned as it contains the hard-coded values.
* While creating a new event, the option of location, start, and end time will be added if an event is not all day.
* Pagination will be implemented in the Organization post and Organization event screen.
* Search functionality will be implemented in the Organization post and Organization event screen.
* Feature to edit created event or post will be implemented.
* Spam notification functionality will be implemented. I.e. whenever a user spams an organization group chats, the notification of the user will be shown in the notification bar.
* Block/Unblock user functionality will be implemented. I.e. whoever user spams the organization group chats will be shown on that page, and organization admins have the power to block/unblock that user.
* Search functionality will be implemented in the Block/Unblock screen.
* Pagination will be implemented in the Block/Unblock screen.
* Multiple screen responsiveness issues will be fixed.

### PR DESCRIPTION
**Pull request by asmitsirohi**
**What kind of change does this PR introduce?**

feature

**Issue Number:**

Fixes #357 

**Did you add tests for your changes?**

Yes

**Snapshots/Videos:**

* Posts screen with pagination and complete dynamic list of posts.
image

* Edit Post Functionality
image

* Search posts functionality
image

* Events screen with pagination and complete dynamic list of events.
image

* Edit Event Functionality
image

* Search event functionality
image

* Spam notification
image

* List of users who spammed in organization group chat
image

* Block user who spammed in organization group chat
image

**If relevant, did you update the documentation?**

No

**Summary**

This pull request includes features, which is required to implement. The complete info is as follows:

* The posts and events list is redesigned as it contains the hard-coded values.
* While creating a new event, the option of location, start, and end time is added if an event is not all day.
* Pagination is implemented in the Organization post and Organization event screen.
* Search functionality is implemented in the Organization post and Organization event screen.
* Feature to edit created event or post is implemented.
* Spam notification functionality is implemented. I.e. whenever a user spams an organization group chats, the notification of the user will be shown in the notification bar.
* Block/Unblock user functionality is implemented. I.e. whoever user spams the organization group chats will be shown on that page, and organization admins have the power to block/unblock that user.
* Search functionality is implemented in the Block/Unblock screen.
* Pagination is implemented in the Block/Unblock screen.
* Multiple screen responsiveness issues are fixed.

**Does this PR introduce a breaking change?**

No

**Other information**

This pull request is under GSoC'22.

**Have you read the [contributing guide](https://github.com/PalisadoesFoundation/talawa-admin/blob/master/CONTRIBUTING.md)?**

Yes

### CODE REVIEW

**Comment by asmitsirohi** [comment]
@yasharth291 Please review

File: src/components/EventListCard/EventListCard.test.tsx [hunk 1]
```diff
@@ -14,11 +17,33 @@ const MOCKS = [
...
-          },
-        ],
+        removeEvent: {
+          _id: '1',
+        },
+      },
+    },
+  },
+  {
+    request: {
```
Comment (yasharth291) [inline src/components/EventListCard/EventListCard.test.tsx:27]: why is this dummy data

File: src/components/OrgPostCard/OrgPostCard.test.tsx [hunk 2]
```diff
@@ -14,11 +17,26 @@ const MOCKS = [
...
-        ],
+        removePost: {
+          _id: '1',
+        },
+      },
+    },
+  },
+  {
+    request: {
+      query: UPDATE_POST_MUTATION,
```
Comment (yasharth291) [inline src/components/OrgPostCard/OrgPostCard.test.tsx:28]: same here

File: src/components/OrgPostCard/OrgPostCard.tsx [hunk 3]
```diff
@@ -15,39 +19,93 @@ interface OrgPostCardProps {
...
+        variables: {
+          id: props.id,
+          title: postformState.posttitle,
+          text: postformState.postinfo,
+        },
+      });
+
+      /* istanbul ignore next */
+      if (data) {
+        window.alert('Post Updated successfully.');
```
Comment (yasharth291) [inline src/components/OrgPostCard/OrgPostCard.tsx:76]: dont user alert here it doesnt look good

File: src/components/plugins/DummyPlugin/DummyPlugin.test.jsx [hunk 4]
```diff
@@ -7,11 +8,13 @@ import DummyPlugin from './DummyPlugin';
 describe('Testing dummy plugin', () => {
   test('should render props and text elements test for the page component', () => {
     const { getByText } = render(
-      <BrowserRouter>
-        <Provider store={store}>
-          <DummyPlugin />
-        </Provider>
-      </BrowserRouter>
+      <MockedProvider>
```
Comment (yasharth291) [inline src/components/plugins/DummyPlugin/DummyPlugin.test.jsx:11]: there is no test written here

File: src/screens/BlockUser/BlockUser.test.tsx [hunk 5]
```diff
@@ -0,0 +1,170 @@
...
+
+const MOCKS = [
+  {
+    request: {
+      query: USER_LIST,
+    },
+    result: {
+      data: {
+        users: [
+          {
```
Comment (yasharth291) [inline src/screens/BlockUser/BlockUser.test.tsx:25]: why dummy data

**Review by yasharth291 — CHANGES_REQUESTED** [review]

File: src/screens/BlockUser/BlockUser.test.tsx (hunk 5 above)
Comment (asmitsirohi) [reply src/screens/BlockUser/BlockUser.test.tsx:25]: This dummy data is to mock graphql queries used in BlockUser.tsx file.

File: src/components/plugins/DummyPlugin/DummyPlugin.test.jsx (hunk 4 above)
Comment (asmitsirohi) [reply src/components/plugins/DummyPlugin/DummyPlugin.test.jsx:11]: In DummyPlugin.tsx file only a single statement is present as you can see in the image: 

image

So, only one assertion is done.

File: src/components/EventListCard/EventListCard.test.tsx (hunk 1 above)
Comment (asmitsirohi) [reply src/components/EventListCard/EventListCard.test.tsx:27]: This dummy data is to mock graphql queries used in EventListCard.tsx file.

File: src/components/OrgPostCard/OrgPostCard.test.tsx (hunk 2 above)
Comment (asmitsirohi) [reply src/components/OrgPostCard/OrgPostCard.test.tsx:28]: This dummy data is to mock graphql queries used in OrgPostCard.tsx file.

File: src/components/OrgPostCard/OrgPostCard.tsx (hunk 3 above)
Comment (asmitsirohi) [reply src/components/OrgPostCard/OrgPostCard.tsx]: JavaScript alert has been replaced with react-toast alert in the last commit.

**Review by yasharth291 — APPROVED** [review]

**Comment by asmitsirohi** [comment]
@palisadoes sir, please merge this PR as @yasharth291 has already approved this.

**Review by palisadoes — APPROVED** [review]

**Pull request merged**